File: Minimum Recolors to Get K Consecutive Black Blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int whiteCount = 0;
        for(int i = 0 ;i < k ; ++i){
            if(blocks[i]=='W'){
                whiteCount++;
            }
        }

        int minimum = whiteCount;

        int size = blocks.size();
        for(int i = k; i < size; ++i){
            if(blocks[i]=='W'){
                whiteCount++;
            }
            if(blocks[i-k]=='W'){
                whiteCount--;
            }
            minimum = min(minimum, whiteCount);
            if(!minimum){
                return minimum;
            }
        }

        return minimum;
    }
};

File: Minimum Recolors to Get K Consecutive Black Blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        size = len(blocks)
        whiteCount = 0
        for a in blocks[:k]:
            if a is 'W':
                whiteCount += 1
        if whiteCount is 0:
            return 0
        minimum = whiteCount
        for i in range(k,size):
            if blocks[i] is 'W':
                whiteCount += 1
            if blocks[i-k] is 'W':
                whiteCount -= 1
            
            minimum = min(minimum, whiteCount)
            if minimum is 0:
                return 0
        
        return minimum
